### DISCUSSION TITLE
Get All possible Enum values

### DISCUSSION THREAD

**Discussion by safaci2000** [go-jet/jet#384] [Q&A] [answered]
There's a few objects that are available that reference enum.  My enum is called Status, so I'll use to refer to it in the body.

- enum package where each individual enum has a var named Status after the enum with a struct that contains all possible values.
- model package, which has a set of const and a new type.  I can use Scan() and String() to construct the enums and get its string equivalent.
- The table package simply references to the enum as its underlying data type ie. postgres.ColumnString.

Now, I don't see any way to get all the possible value of my enum.  I don't think I can override this behavior through the custom generator, but maybe I missed something?

Is this possible with Jet?

**Comment by houtn11** [comment]
No, that's currently not possible. Why do you need it?

**Reply by safaci2000** [reply]
I have a supporting endpoint that drives the UI, so it's handy to be able to just send the entire list rather than enumerating it in the handler.

For example:

	data := []tasksModels.TaskType{tasksModels.TaskType_CtrlColonSubmit, tasksModels.TaskType_CtrlColonStop, tasksModels.TaskType_S3ColonInit, tasksModels.TaskType_PcapColonMerge}

At some point, if I add a new DataType I have to remember to add to the list, which is error prone.  The nice advantage of having a type safe system like Jet or SQLC is that it's aware of all the given types, so it'd be nice to be able to just do a return data := tasksModels.TaskTypes() 

I can of course generate it from a SQL distinct query but that seems unnecessary considering we have everything in code.

**Reply by go-jet** [reply]
Makes sense to add a such a feature.

**Reply by safaci2000** [reply]
https://github.com/go-jet/jet/pull/387 I created a small PR that would address this.  

A few food for thought misc comments.

1. [testcontainers](https://testcontainers.com/) are really really nice and makes it much easier to setup the test framework.
2. it would be nice to static embed the templates and allow the user to alternatively provide different foobar.go.tmpl.  It would allow a lot more flexibility for customization without the need to update the core project. 
ie. Look for templates/foobar.go.tmpl if it doesn't exist use the embedded resource

**Reply by safaci2000** [reply, marked as answer]
master now support this out of the box, so marking this as resolved.
